actors/miner/tests/miner_actor_test_commitment.rs

```diff
@@ -234,19 +240,444 @@ mod miner_actor_test_commitment {
+            expect_abort_contains_message(ExitCode::USR_ILLEGAL_ARGUMENT, "deals too large", ret);
+        }
+
+        // Try to precommit while in fee debt with insufficient balance
+        {
+            let mut st: State = rt.get_state();
+            st.fee_debt =
+                rt.balance.borrow().clone() + TokenAmount::from(10_000_000_000_000_000_000i128);
+            rt.replace_state(&st);
+            let precommit_params =
+                h.make_pre_commit_params(102, challenge_epoch, expiration, vec![]);
+            let ret = h.pre_commit_sector_internal(
+                &mut rt,
+                precommit_params,
+                util::PreCommitConfig::default(),
+                false,
+            );
+            expect_abort_contains_message(
+                ExitCode::USR_INSUFFICIENT_FUNDS,
+                "unlocked balance can not repay fee debt",
+                ret,
+            );
+            // reset state back to normal
+            st.fee_debt = TokenAmount::zero();
+            rt.replace_state(&st);
+            rt.reset();
+        }
+
+        // Try to precommit with an active consensus fault
+        {
+            let st: State = rt.get_state();
+            let fault = ConsensusFault {
+                target: h.receiver,
+                epoch: rt.epoch - 1,
+                fault_type: ConsensusFaultType::DoubleForkMining,
+            };
+            let test_addr = Address::new_actor(b"satoshi");
+            h.report_consensus_fault(&mut rt, test_addr, Some(fault));
+            let precommit_params =
+                h.make_pre_commit_params(102, challenge_epoch, expiration, vec![]);
+            let ret = h.pre_commit_sector_internal(
+                &mut rt,
+                precommit_params,
+                util::PreCommitConfig::default(),
+                false,
+            );
+            expect_abort_contains_message(ExitCode::USR_FORBIDDEN, "active consensus fault", ret);
+            // reset state back to normal
+            rt.replace_state(&st);
+            rt.reset();
+        }
+    }
 
-    #[ignore]
     #[test]
-    fn fails_with_too_many_deals() {}
+    fn fails_with_too_many_deals() {
+        let setup = |proof: RegisteredSealProof| -> (MockRuntime, ActorHarness, DeadlineInfo) {
+            let period_offset = ChainEpoch::from(100);
+
+            let mut h = ActorHarness::new(period_offset);
+            h.set_proof_type(proof);
+            let mut rt = h.new_runtime();
+            rt.set_balance(TokenAmount::from(BIG_BALANCE));
+            rt.set_received(TokenAmount::zero());
+
+            rt.set_epoch(period_offset + 1);
+            h.construct_and_verify(&mut rt);
+            let deadline = h.deadline(&rt);
+            (rt, h, deadline)
+        };
+
+        let make_deal_ids = |n| -> Vec<DealID> { (0..n).collect() };
+
+        let sector_number: SectorNumber = 100;
+        let deal_limits = [
+            (RegisteredSealProof::StackedDRG2KiBV1P1, 256),
```

Comment:
I don't understand how 2k sectors are working with this test.  The go tests explicitly monkeypatch the allowed window posts types policy map for this test and do the same for allowed commit types map globally for the miner_test package.  How are 2k operations getting past checks at construction and precommit in this test?

Reply:
I'm guessing something about the `#[cfg(feature = "sector-2k")]` being set for test running?